File: app/src/main/java/br/com/stonesdk/sdkdemo/activities/DisconnectPinpadActivity.java
package br.com.stonesdk.sdkdemo.activities;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.Spinner;

import java.util.ArrayList;
import java.util.List;

import br.com.stonesdk.sdkdemo.R;
import stone.utils.Stone;

public class DisconnectPinpadActivity extends AppCompatActivity {

    Spinner pinpadsSpinner;
    Button disconnectButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_disconnect_pinpad);

        pinpadsSpinner = findViewById(R.id.pinpadsSpinner);
        disconnectButton = findViewById(R.id.disconnectButton);

        setPinpadsToSpinner();

        disconnectButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                int pinpadSelected = pinpadsSpinner.getSelectedItemPosition();
                Stone.removePinpadAtIndex(Stone.getPinpadFromListAt(pinpadSelected));
                setPinpadsToSpinner();
            }
        });
    }

    private void setPinpadsToSpinner() {

        List<String> pinpads = new ArrayList<>();
        for (Integer i = 0; i < Stone.getPinpadListSize(); i++)
            pinpads.add(Stone.getPinpadFromListAt(i).getName());


        ArrayAdapter<String> adapter = new ArrayAdapter<>(this, android.R.layout.simple_spinner_item, pinpads);
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        pinpadsSpinner.setAdapter(adapter);

    }
}

File: app/src/main/java/br/com/stonesdk/sdkdemo/activities/DisconnectPinpadActivity.kt
package br.com.stonesdk.sdkdemo.activities

import android.os.Bundle
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.Spinner
import androidx.appcompat.app.AppCompatActivity
import br.com.stonesdk.sdkdemo.databinding.ActivityDisconnectPinpadBinding
import stone.utils.Stone

class DisconnectPinpadActivity : AppCompatActivity() {

    private lateinit var binding: ActivityDisconnectPinpadBinding
    
    var pinpadsSpinner: Spinner? = null
    var disconnectButton: Button? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        
        
        super.onCreate(savedInstanceState)
        binding = ActivityDisconnectPinpadBinding.inflate(layoutInflater)
        setContentView(binding.root)

        pinpadsSpinner = binding.pinpadsSpinner
        disconnectButton = binding.disconnectButton

        setPinpadsToSpinner()

        binding.disconnectButton.setOnClickListener {
            val pinpadSelected = binding.pinpadsSpinner.selectedItemPosition
            Stone.removePinpadAtIndex(Stone.getPinpadFromListAt(pinpadSelected))
            setPinpadsToSpinner()
        }
    }

    private fun setPinpadsToSpinner() {
        val pinpads: MutableList<String> = ArrayList()
        for (i in 0 until Stone.getPinpadListSize()) pinpads.add(Stone.getPinpadFromListAt(i).name)


        val adapter = ArrayAdapter(this, android.R.layout.simple_spinner_item, pinpads)
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        pinpadsSpinner!!.adapter = adapter
    }
}
